uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -1364,4 +1365,78 @@ private static function getLeaveRequestBalance($params) {
 
     return self::calculateBalanceChangeFromCreateParams($params);
   }
+
+  /**
+   * There are some fields that are only required according to the calculation
+   * unit of the Leave Request's Absence Type. This method makes sure that the
+   * non-required fields according to the calculation unit will be reset.
+   *
+   * When the calculation unit is hours, the from_date_type and to_date_type
+   * fields are not required, while the from_date_amount and to_date_amount are.
+   * When the calculation unit is in days, the from_date_amount and to_date_amount
+   * are required and the from_date_type and to_date_type are not.
+   *
+   * @param array $params
+   *
+   * @return mixed
+   * @throws \Exception
```

Comment:
I think we should decide on whether to include these or not, I tried to do a poll on it and it was 2 votes against vs 0 votes for

Reply:
I think it's not as simple as not using them or not, but I removed it for now